scripts/mkdata.py

```diff
@@ -105,125 +65,200 @@ def gen_r_script(data, layout, single_eas, out_dir):
      at = seq(1 , 5 , 1),
       labels = c('(a)','(b)','(c)','(d)','?'),
      tick=T , cex=0.3)
-abline(v=2.5,lty=1, col="grey")
+abline(v=${line}.5,lty=1, col="grey")
 
-lvl <- levels(data$sample)
-props <- summary(data$sample) / nrow(data)
+lvl <- levels(data$$sample)
+props <- summary(data$$sample) / nrow(data)
 
 for (i in 1:length(lvl)) {{
     l <- lvl[i]
-    v <- data[ data$sample==l, "values" ]
+    v <- data[ data$$sample==l, "values" ]
     j <- jitter(rep(i, length(v)), amount=props[i]/2)
     points(j, v, pch=20, col=rgb(0,0,0,0.7))
 }}
 """
 
-    script += f"# Total n: {ns} (should be 1024)\n"
-    script += "dev.off()\n"
-    out_dir.mkdir(parents=True, exist_ok=True)
-    with open(out_dir / "plot.R", "w") as stream:
-        stream.write(script)
+R_SCRIPT_FOOTER_TEMPLATE = """
+# Total n: ${ns} (should be 1024)
+dev.off()\n
+"""
 
 
-def gen_stats(data):
-    text = r"""
+LATEX_HEADER_TEMPLATE = r"""
 \begin{tabular}{ |c|r|r|r||r|r|r| }\hline
-   & \multicolumn{3}{c||}{\textbf{Core 1}} & \multicolumn{3}{c|}{\textbf{Core 2}} \\\hline
+   & \multicolumn{3}{c||}{\textbf{Core ${core0}}} & \multicolumn{3}{c|}{\textbf{Core ${core1}}} \\\hline
   \textbf{EA} & \textbf{max(a)} \textit{(ms)} & \textbf{max(b)} \textit{(ms)} & %
-  $\bm{R(a, b)}$ \textit{(\%)}& %
+  $$\bm{R(a, b)}$$ \textit{(\%)}& %
   \textbf{max(c)} \textit{(ms)} & \textbf{max(d)} \textit{(ms)} & %
-  $\bm{R(c, d)}$ \textit{(\%)} \\\hline
+  $$\bm{R(c, d)}$$ \textit{(\%)} \\\hline
+"""
+
+LATEX_FOOTER = r"""
+\hline
+\end{tabular}
 """
+
+
+def check_layout(layout):
+    rows = len(layout)
+    assert rows > 0
+    cols = len(layout[0])
+    for row in layout:
+        assert len(row) == cols
+    return rows, cols
+
+def getopts(argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--kdbv", type=Path, required=True)
+    parser.add_argument("--kcfg", type=Path, required=True)
+    parser.add_argument("--kapp", type=Path, required=True)
+    parser.add_argument("--c0-off", type=Path, required=True)
+    parser.add_argument("--c0-on", type=Path, required=True)
+    parser.add_argument("--c1-off", type=Path, required=False)
+    parser.add_argument("--c1-on", type=Path, required=False)
+    parser.add_argument("--output-dir", "-o", type=Path, required=True)
+    parser.add_argument("--task", choices=["G", "H", 'U'], required=True)
+    parser.add_argument("--timer", type=float, default=5e6)
```

Comment:
I think you should remove the default value. This has too big of an impact on the results.
Make it required as well, to force the user to write its timer frequency.